### DISCUSSION TITLE
How to save the results from an multiple alignment

### DISCUSSION THREAD

**Discussion by rpestana94** [biotite-dev/biotite#326] [Q&A] [answered]
I have as test 13 sequences each one about 1270 amino acids in length, and I want to save the alignment result in a file (text, clustal, msf, etc) but I can't find a way to properly do it. 

I have the following code:

```
import biotite.sequence as seq
import biotite.sequence.io.fasta as fasta
import biotite.sequence.graphics as graphics
import biotite.sequence.align as align
import matplotlib.pyplot as plt

fasta_file = fasta.FastaFile.read("test_few_sample.fasta")
all_sequences = fasta.get_sequences(fasta_file)

ids = []
sequences = []

for header, seq_str in fasta_file.items():
    identifier = header.split("|")[-1].split()[0].split(":")[0]
    ids.append(identifier)
    sequences.append(seq.ProteinSequence(seq_str))

matrix = align.SubstitutionMatrix.std_protein_matrix()
alignment, order, tree, distances = align.align_multiple(
    sequences, matrix, gap_penalty=(-10,-1), terminal_penalty=False
)

alignment = alignment[:, order]
ids = [ids[i] for i in order]
```

Thanks in advance

**Comment by padix-key** [comment, marked as answer]
If you would like to save the alignment as fasta file, you can use `biotite.sequence.io.fasta.set_alignment()` (https://www.biotite-python.org/apidoc/biotite.sequence.io.fasta.set_alignment.htm):
```python
fasta_file = fasta.FastaFile()
fasta.set_alignment(fasta_file, alignment, seq_names=ids)
fasta_file.write("alignment.fasta")
```

If you prefer another format, you have to write the file yourself, unfortunately. However, you can start with `Alignment.get_gapped_sequences()` (https://www.biotite-python.org/apidoc/biotite.sequence.align.Alignment.html#biotite.sequence.align.Alignment.get_gapped_sequences) to convert the alignment into gapped sequence strings.

**Reply by rpestana94** [reply]
Thanks
